controllers/machinedeployment_controller.go

```diff
@@ -152,10 +152,6 @@ func (r *MachineDeploymentReconciler) reconcile(_ context.Context, cluster *clus
 	d.Spec.Selector.MatchLabels[clusterv1.ClusterLabelName] = d.Spec.ClusterName
 	d.Spec.Template.Labels[clusterv1.ClusterLabelName] = d.Spec.ClusterName
```

Comment:
Not directly related, but might be good to track a followup item to investigate if we should move the handling of ensure the label matching for the ClusterName is present to the Defaulting webhook as well...

Reply:
tracking: https://github.com/kubernetes-sigs/cluster-api/issues/2302
marked as good first issue, let me know if that's not suitable. :D